listenbrainz/labs_api/labs/api/recording_from_recording_mbid.py

```diff
@@ -90,43 +90,40 @@ def fetch(self, params, offset=-1, count=-1):
                 args = [tuple([psycopg2.extensions.adapt(p) for p in mbids])]
                 curs.execute(query, tuple(args))
 
-                # Build an index of all the fetched recordings and clean up UUID() -> str
+                # Build an index of all the fetched recordings
                 recording_index = {}
                 while True:
                     row = curs.fetchone()
                     if not row:
                         break
 
-                    r = dict(row)
-                    mbid = str(r['recording_mbid'])
-                    r['recording_mbid'] = mbid
-                    recording_index[mbid] = dict(row)
+                    recording_index[row['recording_mbid']] = dict(row)
 
-                # Finally collate all the results, ensuring that we have one entry with original_recording_mbid for each 
+                # Finally collate all the results, ensuring that we have one entry with original_recording_mbid for each
                 # input argument
                 output = []
                 for p in params:
-                    mbid = p['[recording_mbid]'] 
+                    mbid = p['[recording_mbid]']
                     try:
                         r = copy.copy(recording_index[mbid])
                     except KeyError:
                         try:
                             r = copy.copy(recording_index[redirect_index[mbid]])
                         except KeyError:
-                            out = {} 
-                            for k in self.outputs():
-                                if k == 'original_recording_mbid':
-                                    out['original_recording_mbid'] = mbid
-                                else:
-                                    out['original_recording_mbid'] = None
+                            out = dict.fromkeys(self.outputs(), None)
+                            out['original_recording_mbid'] = mbid
                             output.append(out)
                             continue
 
-                    r['[artist_credit_mbids]'] = list(set([str(r) for r in r['artist_credit_mbids']]))
+                    r['[artist_credit_mbids]'] = list(set([r for r in r['artist_credit_mbids']]))
```

Comment:
don't need the list comprehension now that we don't need `str`? `list(set(r['artist_credit_mbids']))`